src/python/pants/backend/native/subsystems/binaries/llvm.py

```diff
@@ -76,17 +77,17 @@ def _filemap(self, all_components_list):
   def path_entries(self):
     return self._filemap([('bin',)])
 
-  def linker(self, platform):
+  def linker(self, platform: Platform) -> Linker:
     return Linker(
       path_entries=self.path_entries,
       exe_filename=platform.match({
         Platform.darwin: "ld64.lld",
         Platform.linux: "lld",
       }),
-      library_dirs=[],
```

Comment:
Thank you for fixing this!! Could you please add a `TODO` here pointing to #5663 and mentioning that this is currently dead code? Sorry for not doing this myself!!